Steps/PrGate.yml

```diff
@@ -112,6 +116,10 @@ steps:
 - ${{ if eq(parameters.install_tools, true) }}:
   - template: InstallCoverageTools.yml
 
+# Potential Extra steps
+- ${{ if eq(parameters.self_host_agent, false) }}:

```

Comment:
Is it a general rule that self-hosted agents do not potentially need extra install steps?

Reply:
That is the current approach. Until when we switch to a full container approach, I think self-hosted agents should meet the prerequisites of tooling. (other than pip modules as they are too dynamic).

Reply:
If that's the case, what do you think about raising an error if the caller has `parameters.self_host_agent` set to `true` and `parameters.extra_install_step` is non-empty?

Reply:
The `parameters.extra_install_step` is not set per matrix entry after the change, but instead being used for each `jobs`. But the `parameters.self_host_agent` is tied to each matrix entry.

This was done because I think it is undesirable to duplicate the pipelines just for selfhost builds, because those will be manual process and hard to manage. Expanding the matrix would be an easier path for repos to enroll if they want.

On the other hand, because the wrapper yaml file (the one that will be synced) contains `resources`, we can only use `extends` instead of `jobs -template` to apply it. This limited the capability of using that file as a template twice in one place.

Reply:
Synced offline on this and added warning message to the `Stesp/PrGate.yml` to make skipped steps more obvious for easier troubleshooting.